packages/common/src/types/metricQuery.ts

```diff
@@ -52,6 +52,9 @@ export type MetricQuery = {
     tableCalculations: TableCalculation[]; // calculations to append to results
     additionalMetrics?: AdditionalMetric[]; // existing metric type
     customDimensions?: CustomDimension[];
+    metadata?: {
```

Comment:
mm, instead of adding a new metadata property on the request, I think we can just send the metric query and recalculate the `hasDateDimension` from the metricQuery in the backend. But this is personal opinion.

Reply:
In order to recalculate the `hasDateDimension` we would have to getExplore(...), which can be heavy and reduce interactivity. Doing it on the BE allows us to use the existing explore (because we query it already on `getChartAndResults` )